### PR TITLE
[12.0][FIX/IMP] hr_timesheet_sheet_period: sudo and default behaviour

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Addressing https://github.com/OCA/timesheet/issues/288

@mreficent @aheficent @astirpe

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
@pedrobaeza a tiny slice of your time is needed regarding the migration, again

**Review by AaronHForgeFlow — COMMENTED** [review]
I missed the field sheet_range in the company view, otherwise how can we set it?

Probably not due to your changes, but it seems the hr_period is not compute correctly in the timesheet. I created monthly periods for 2019 and when I ccreate a timesheet the period is not added. 

Also I consider the hr period field should be hidden in the timesheet in case the period is not set. For sure, this is not due to this Pull Request. Could you also do that? If not, I will create a PR to your branch. Thanks

**Comment by alexey-pelykh** [comment]
@aheficent `sheet_range` is exposed in Settings, as per readme?

**Comment by AaronHForgeFlow** [comment]
Opps yes, please dismiss that comment. 

Regarding the other comment, tt is not big deal, but could you put `invisible="[('hr_period', '=', False)]"` in the view?

**Comment by alexey-pelykh** [comment]
@aheficent added the invisible tag, thanks. Regarding the period issue - I'll take a look

**Comment by alexey-pelykh** [comment]
Looked briefly into code, maybe because it's not required? Wild guess, just add the value at create()

**Comment by AaronHForgeFlow** [comment]
The field is readonly, but it is only filled by a default method. I wonder if it should be a computed field, as it is for the field fiscal_year in model account.move.line

**Comment by alexey-pelykh** [comment]
@aheficent I did some more amendments and did a functional test, seems to me all good. Please re-review

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:  Thanks for the work. It works fine now.

**Comment by alexey-pelykh** [comment]
Who else would be appropriate to review this?

**Comment by AaronHForgeFlow** [comment]
cc @mreficent  Could you please review?

File: hr_timesheet_sheet_period/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -1,2 +1,10 @@
-Create first the Payroll Fiscal Years and Payroll
-Periods from 'Human Resources > Configuration > Payroll'
+Configure Timesheet Sheets:
```
Comment (MiquelRForgeFlow) [inline hr_timesheet_sheet_period/readme/CONFIGURE.rst]:
```suggestion
Configure *Timesheet Sheets*:
```

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

File: hr_timesheet_sheet_period/readme/CONFIGURE.rst (hunk 1 above)
Comment (alexey-pelykh) [reply hr_timesheet_sheet_period/readme/CONFIGURE.rst]: 👍

**Comment by alexey-pelykh** [comment]
@pedrobaeza one brief look please on migrations

**Comment by pedrobaeza** [comment]
I don't know the context, so not able to say

**Comment by alexey-pelykh** [comment]
@pedrobaeza context: whomever had this module 12.0.1.0.0 installed and upgrades to 12.0.2.0.0 should get `sheet_range` field of `res.company` set to `100`

**Comment by alexey-pelykh** [comment]
(since before just mere presence of this module was treated as override of any `sheet_range` field value)

**Comment by pedrobaeza** [comment]
That migration script part seems OK then

**Pull request merged**